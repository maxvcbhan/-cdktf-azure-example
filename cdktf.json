{
	"language": "typescript",
	"app": "npx ts-node main.ts",
	"projectId": "9c767737-300f-49c8-8c49-4152b75a4447",
	"sendCrashReports": "false",
	"terraformProviders": [],
	"terraformModules": [],
	"context": {
		"excludeStackIdFromLogicalIds": "true",
		"allowSepCharsInLogicalIds": "true"
	}
}
