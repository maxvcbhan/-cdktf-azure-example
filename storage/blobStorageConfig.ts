import { Construct } from 'constructs';
import {
	StorageAccountConfig,
} from '@cdktf/provider-azurerm/lib/storage-account';
import {
	StorageAccount,
	StorageBlob,
	StorageContainer,
} from '@cdktf/provider-azurerm';
import { ResourceGroup } from '@cdktf/provider-azurerm/lib/resource-group';


export class BlobStorageConfig extends Construct {
	public readonly container: StorageContainer;
	public readonly storageAccount: StorageAccount;
	public readonly storageConfig: StorageAccountConfig;
	public readonly storageBlob: StorageBlob;

	constructor(scope: Construct, name: string,
				storageAccountName: string,
				baseNameApplication: string,
				resourceGroup: ResourceGroup,
	) {
		super(scope, name);

		this.storageConfig = {
			accountTier: 'Standard',
			resourceGroupName: resourceGroup.name,
			name: storageAccountName,
			location: resourceGroup.location,
			accountReplicationType: 'LRS',
		};
		this.storageAccount = new StorageAccount(
			this,
			`${baseNameApplication}-storage-account`,
			this.storageConfig,
		);
		this.container = new StorageContainer(
			this,
			`${baseNameApplication}-blob-container`,
			{
				name: `${baseNameApplication}-container`,
				containerAccessType: 'container',
				storageAccountName: this.storageAccount.name,
			},
		);
		this.storageBlob = new StorageBlob(this, `${baseNameApplication}-blob`, {
			accessTier: 'Cool',
			storageAccountName: this.storageAccount.name,
			name: 'document',
			storageContainerName: this.container.name,
			type: 'Block',
		});
	}
}
