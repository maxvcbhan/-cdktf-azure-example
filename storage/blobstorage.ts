import { Construct } from 'constructs';
import {
	StorageAccountConfig,
} from '@cdktf/provider-azurerm/lib/storage-account';
import {
	StorageAccount,
	StorageBlob,
	StorageContainer,
} from '@cdktf/provider-azurerm';
import { ResourceGroup } from '@cdktf/provider-azurerm/lib/resource-group';


export class BlobStorage extends Construct {

	constructor(scope: Construct, name: string,
				storageAccountName: string,
				resourceGroup: ResourceGroup,
				baseNameApplication: string,
	) {
		super(scope, name);

		const storageConfig: StorageAccountConfig = {
			accountTier: 'Standard',
			resourceGroupName: resourceGroup.name,
			name: storageAccountName,
			location: resourceGroup.location,
			accountReplicationType: 'LRS',
		};
		const sa = new StorageAccount(
			this,
			`${baseNameApplication}-storage-account`,
			storageConfig,
		);
		const container: StorageContainer = new StorageContainer(
			this,
			`${baseNameApplication}-blob-container`,
			{
				name: `${baseNameApplication}-container`,
				containerAccessType: 'container',
				storageAccountName: sa.name,
			},
		);
		new StorageBlob(this, `${baseNameApplication}-blob`, {
			accessTier: 'Cool',
			storageAccountName: sa.name,
			name: 'document',
			storageContainerName: container.name,
			type: 'Block',
		});
	}
}
