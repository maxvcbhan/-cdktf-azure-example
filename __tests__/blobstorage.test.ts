import 'cdktf/lib/testing/adapters/jest'; // Load types for expect matchers
import { Testing } from 'cdktf';
import { BlobStorageConfig } from '../storage/blobStorageConfig';
import {
	ResourceGroup,
	StorageAccount, StorageBlob,
	StorageContainer,
} from '@cdktf/provider-azurerm';

const dotenv = require('dotenv');
dotenv.config({ path: '.development.env' });
console.log(process.env);
describe('My CDKTF Application', () => {

	describe('Block storage', () => {
		const stack = Testing.synthScope((scope) => {
			new BlobStorageConfig(scope, 'my-app-under-test', 'mock-storage-account-name', 'mock-base-app',
				new ResourceGroup(scope, 'acc', {
					location: 'mock-location', name: 'mock-name',
				}));
		});

		it('should contain a storage account', () => {
			expect(stack).toHaveResourceWithProperties(StorageAccount, {
				account_replication_type: 'LRS',
				account_tier: 'Standard',
			});
		});
		it('should have storage container', () => {
			expect(stack).toHaveResourceWithProperties(StorageContainer, {
				container_access_type: 'container',
			});
		});
		it('should have storage blob', () => {
			expect(stack).toHaveResourceWithProperties(StorageBlob, {
				access_tier: 'Cool',
				type: 'Block',
			});
		});
	});
});
describe('Unit testing using snapshots', () => {
	it('Tests the blob storage snapshot', () => {
		expect(
			Testing.synthScope((stack) => {
				new BlobStorageConfig(stack, 'my-app-under-test', 'mock-storage-account-name', 'mock-base-app',
					new ResourceGroup(stack, 'acc', {
						location: 'mock-location', name: 'mock-name',
					}));

			})).toMatchInlineSnapshot(`
"{
  \\"resource\\": {
    \\"azurerm_resource_group\\": {
      \\"acc\\": {
        \\"location\\": \\"mock-location\\",
        \\"name\\": \\"mock-name\\"
      }
    },
    \\"azurerm_storage_account\\": {
      \\"my-app-under-test_mock-base-app-storage-account_A4E8D618\\": {
        \\"account_replication_type\\": \\"LRS\\",
        \\"account_tier\\": \\"Standard\\",
        \\"location\\": \\"\${azurerm_resource_group.acc.location}\\",
        \\"name\\": \\"mock-storage-account-name\\",
        \\"resource_group_name\\": \\"\${azurerm_resource_group.acc.name}\\"
      }
    },
    \\"azurerm_storage_blob\\": {
      \\"my-app-under-test_mock-base-app-blob_B52E41CB\\": {
        \\"access_tier\\": \\"Cool\\",
        \\"name\\": \\"document\\",
        \\"storage_account_name\\": \\"\${azurerm_storage_account.my-app-under-test_mock-base-app-storage-account_A4E8D618.name}\\",
        \\"storage_container_name\\": \\"\${azurerm_storage_container.my-app-under-test_mock-base-app-blob-container_E605FBD8.name}\\",
        \\"type\\": \\"Block\\"
      }
    },
    \\"azurerm_storage_container\\": {
      \\"my-app-under-test_mock-base-app-blob-container_E605FBD8\\": {
        \\"container_access_type\\": \\"container\\",
        \\"name\\": \\"mock-base-app-container\\",
        \\"storage_account_name\\": \\"\${azurerm_storage_account.my-app-under-test_mock-base-app-storage-account_A4E8D618.name}\\"
      }
    }
  }
}"
`);
	});
});


