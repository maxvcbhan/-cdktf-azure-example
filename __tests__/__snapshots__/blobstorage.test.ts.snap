// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit testing using snapshots Tests the snapshot 1`] = `
"{
  \\"provider\\": {
    \\"azurerm\\": [
      {
        \\"features\\": {
        },
        \\"subscription_id\\": \\"mock-subscription-id\\"
      }
    ]
  },
  \\"terraform\\": {
    \\"required_providers\\": {
      \\"azurerm\\": {
        \\"source\\": \\"azurerm\\",
        \\"version\\": \\"3.12.0\\"
      }
    }
  }
}"
`;
